### PR TITLE
Speed up data conversion with pyarrow operations

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:*

*Description of changes:*
- Rewrite `generate_univariate_targets_from_multivariate` using pyarrow table operations instead of `dataset.map` with Python loops (10x faster)
- Rewrite `PandasAdapter._to_long_df` to build flat pyarrow table then `to_pandas()`, avoiding `np.repeat` on strings (10x faster)
- Fix `combine_univariate_predictions_to_multivariate` to use `table.take()` instead of `Dataset.select()` to avoid lazy `_indices` views
- `convert_input_data` per window: 318ms -> 94ms (e.g. saving 6s of runtime on 1 boomlet task)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fev/adapters.py [hunk 1]
```diff
@@ -58,21 +58,21 @@ class PandasAdapter(DatasetAdapter):
...
-                df_dict[col] = np.concatenate(df[col])
-        return pd.DataFrame(df_dict).astype({id_column: str})
+        """Convert time series dataset into long DataFrame format via pyarrow."""
+        import pyarrow as pa
+        import pyarrow.compute as pc
+
+        if getattr(dataset, "_indices", None) is not None:
+            dataset = dataset.flatten_indices()
+        table = dataset.data.table
+        seq_cols = [col for col in table.column_names if col != id_column]
```
Comment (abdulfatir) [inline src/fev/adapters.py:68]: ~~Have all static cols been removed already?~~ Nevermind, they have been removed.

File: src/fev/utils.py [hunk 2]
```diff
@@ -229,27 +228,6 @@ def generate_fingerprint(dataset: datasets.Dataset, num_rows_to_check: int = 3)
...
-                if key not in generate_univariate_targets_from:
-                    value = batch[key][i]
-                    if key == id_column:
-                        value = value + "_" + target_col
-                    expanded_batch[key].append(value)
-            expanded_batch[target_column].append(batch[target_col][i])
-    return dict(expanded_batch)
-
-
 def generate_univariate_targets_from_multivariate(
```
Comment (abdulfatir) [inline src/fev/utils.py:231]: Should we add tests for this?

**Review by abdulfatir — APPROVED** [review]
Awesome! Thanks @shchur. Looks great. I am wondering if we should add tests for some of these things either comparing the efficient implementation with datasets version or simply ensuring that we get the expected results and there are no weird edge cases?

File: src/fev/utils.py (hunk 2 above)
Comment (shchur) [reply src/fev/utils.py:231]: Good idea, I've added a few test cases
